client/src/partials/patients/registry/registry.js

```diff
@@ -14,60 +14,68 @@ PatientRegistryController.$inject = [
+    headerCellFilter: 'translate',
     columnDefs : [
-      { field : 'reference', name : $translate.instant('TABLE.COLUMNS.REFERENCE') },
-      { field : 'patientName', name : $translate.instant('TABLE.COLUMNS.NAME') },
-      { field : 'patientAge', name : $translate.instant('TABLE.COLUMNS.AGE') },
-      { field : 'sex', name : $translate.instant('TABLE.COLUMNS.GENDER') },
-      { field : 'hospital_no', name : $translate.instant('TABLE.COLUMNS.HOSPITAL_FILE_NR') },
-      { field : 'registration_date', cellFilter:'date', name : $translate.instant('TABLE.COLUMNS.DATE_REGISTERED') },
-      { field : 'last_visit', cellFilter:'date', name : $translate.instant('TABLE.COLUMNS.LAST_VISIT') },
-      { field : 'dob', cellFilter:'date', name : $translate.instant('TABLE.COLUMNS.DOB') }  
+      { field : 'reference', displayName : 'TABLE.COLUMNS.REFERENCE', headerCellFilter: 'translate' },
+      { field : 'patientName', displayName : 'TABLE.COLUMNS.NAME', headerCellFilter : 'translate' },
+      { field : 'patientAge', displayName : 'TABLE.COLUMNS.AGE', headerCellFilter : 'translate'  },
+      { field : 'sex', displayName : 'TABLE.COLUMNS.GENDER', headerCellFilter : 'translate'  },
+      { field : 'hospital_no', displayName : 'TABLE.COLUMNS.HOSPITAL_FILE_NR', headerCellFilter : 'translate'  },
+      { field : 'registration_date', cellFilter:'date', displayName : 'TABLE.COLUMNS.DATE_REGISTERED', headerCellFilter : 'translate'  },
+      { field : 'last_visit', cellFilter:'date', displayName : 'TABLE.COLUMNS.LAST_VISIT', headerCellFilter : 'translate'  },
+      { field : 'dob', cellFilter:'date', displayName : 'TABLE.COLUMNS.DOB', headerCellFilter : 'translate'  },
+      { name : 'Actions', displayName : '', cellTemplate : '<div style="padding : 5px"><a ui-sref="patientRecord({patientID : row.entity.uuid})"><span class="glyphicon glyphicon-list-alt"></span> Record</a> <a ui-sref="patientEdit({uuid : row.entity.uuid})"><span class="glyphicon glyphicon-edit"></span> Edit</a></div>'}
     ],
     enableSorting : true
   };
-
+  
   // load Patient Registry Grid
   function loadGrid() {
+    vm.loading = true;
     Patients.read().then(function (patients) {
+      vm.loading = false;
       patients.forEach(function (patient) {
         patient.patientAge = momentAge(patient.dob); 
       });      
       vm.uiGridOptions.data = patients;
     });
   }
 
-  // called on modules start
-  function startup() {
-    loadGrid();
-  }
-
   // Search and filter data in Patiens Registry
   function search() {
+    vm.loading = true; 
     Patients.openSearchModal()
     .then(function (data) {
       var response = data.response;
-      vm.filters = data.filters
+      vm.loading = false;
+      vm.filters = data.filters;
       response.forEach(function (patient) {
         patient.patientAge = momentAge(patient.dob); 
       });      
       vm.uiGridOptions.data = response;
+    })
+    .catch(function () { 
+        vm.loading = false;
```

Comment:
Instead of repeating yourself, could you use the `.finally()` block to toggle the loading indicator off?

```js
Patients.openSearchModal()
.then( /* code ... */ )
.catch( /* code ... */)
.finally(function () { 
  vm.loading = false;
});
```